api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -85,15 +85,15 @@ public static SpanContext createFromRemoteParent(
    * @return the trace identifier associated with this {@code SpanContext}.
    * @since 0.1.0
    */
-  public abstract TraceId getTraceId();
+  public abstract CharSequence getTraceId();
```

Comment:
Would name this getBase16TraceId.

I would also add helpers for propagators copyBytesTraceId copyBase16TraceId similar to current methods on TraceId.

Same applies to all the components.